sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -41,11 +41,11 @@ public override void OnSendingRequest(HttpMessage message)
         {
             base.OnSendingRequest(message);
 
-            // Add a x-ms-date header if it doesn't already exist
-            if (IncludeXMsDate && !message.Request.Headers.Contains(Constants.HeaderNames.Date))
+            // Add a x-ms-date header
+            if (IncludeXMsDate)
             {
                 var date = DateTimeOffset.UtcNow.ToString("r", CultureInfo.InvariantCulture);
-                message.Request.Headers.Add(Constants.HeaderNames.Date, date);
+                message.Request.Headers.SetValue(Constants.HeaderNames.Date, date);
```

Comment:
This concerns me a little bit because it's a subtle semantic change to GA'ed functionality.  @seanmcc-msft is there any chance a customer might be setting their own `Date` or `x-ms-date` _(maybe adding a slight delay if they know it'll be held in a queue for processing before we send it)_ that this would overwrite?  Was there a specific reason the Track 1 code did things this way?  [See the docs](https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key#specifying-the-date-header) as well for additional context.

Reply:
@tg-msft, I doubt customers are manually setting the x-ms-date header.  I don't think there's a specify reason for Track 1 to have this functionality, there's a good chance it was an oversight.

Reply:
I checked with @alzimmermsft and Java sets the date per retry.